integrations/webchat/agent-assist/agent-verify/README.md

```diff
@@ -0,0 +1,188 @@
+
+
+### Step 4: Go to the Integrations page from the Admin panel
+
+![Genesys Admin Page](images/admin-page.png)
+
+From the Admin panel click on the `Integrations` link to create a new integration.
+
+### Step 5: Create a new integration
+
+![Genesys Integrations Page](images/integrations-page.png)
+
+Add a new integration by clicking the + sign in the top right of the page.
+
+### Step 6: Install the Interaction Widget
+
+![Genesys Interaction Widget Page](images/interaction-widget-page.png)
+
+The Interaction Widget is needed to embed third-party webapps via an iframe in the Agent UI.
+
+### Step 7: Configure the Interaction Widget
+
+![Genesys Interaction Widget Config Page](images/interaction-widget-config-page.png)
+
+After you provide a name for your widget on the Details tab, you will need to navigate to the Configuration tab. From here you will need to configure both the `Application URL`, the `Iframe Sandbox Options` and `Group Filtering`.
+
+#### Application URL
+The `Application URL` should take the following form:
+
+```
+http://<webserver>/genesys-agent-assist-verify.html?clientID=<genesys client id>&environment=mypurecloud.com&conversationID={{pcConversationId}}
+```
+`webserver`: host address and port for the webserver that is serving up the client application files included in this sample application.
+
+`clientID`: generated when you create the OAuth credentials to be used for the implicit token login (see below).
+
+`environment`: based on the location of your Genesys account. The environment shown works for North American accounts.
+
+`conversationID`: generated by Genesys at runtime and the field will be populated by the agent application.
+
+#### Iframe Sandbox Options
+The `Iframe Sandbox Options` should be filled in with these options:
+
+```
+allow-scripts,allow-same-origin,allow-forms,allow-modals,allow-popups,allow-presentation,allow-downloads
+```
+#### Group Filtering
+The `Group Filtering` section is selected from a pre-configured group that can be created from the Admin page under Directory->Groups. You need to make sure that all agents that require the new agent-verify widget are listed within the chosen group.
+
+### Step 8: Configure OAuth for the Interaction Widget
+
+![Genesys Interaction Widget Config Page](images/oauth-client-page.png)
+Start by going to `OAuth` under Integrations and it `Add Client` to create a new OAuth client.
+
+![Genesys Interaction Widget Config Page](images/oauth-client-details-top-page.png)
+Here you'll need to configure the App Name associated with the credentials, the Grant Type which must be `Token Implicit Grant (Browser)`, the redirect URL which is shown below:
+```
+http://localhost:8080/genesys-agent-assist-verify.html?clientID=xxxxx&environment=mypurecloud.com
+```
+and the Scope which is set to `notifications`. Note that the redirect URL does not need to include the conversationID.
```

Comment:
"does not need" really should be "must not contain". You cannot put the conversation ID in the redirect URL. I have discovered the redirect URLs must match exactly, including any query parameters.

Reply:
FYI, one trick you can use to hide sensitive information in your screenshots is that when you're using the development tools in your web browser you can actually directly edit the HTML. So you could just edit the HTML that shows the client secret and fill in some bogus data and then you don't have to try to mask it in the screenshot.